smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -683,7 +684,9 @@ private void writeResolvedPath(
         List<HttpBinding> labelBindings = bindingIndex.getRequestBindings(operation, Location.LABEL);
 
         // Always write the bound path, but only the actual segments.
-        writer.write("let resolvedPath = $S;", "/" + trait.getUri().getSegments().stream()
+        writer.write("let resolvedPath = `$L` + $S;",
+                "${basePath?.endsWith('/') ? basePath.slice(0, -1) : (basePath || '')}",
```

Comment:
Is there a unit test to add tests for these cases?

Reply:
Not that I can see, no.

Reply:
Looks like protocol tests let you specify a custom endpoint, so it should be testable there.

Reply:
https://github.com/awslabs/smithy/pull/845